### PR TITLE
[WIP] Check Proxy hash before execution

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-relayer/lib/utils/errors.ts [hunk 1]
```diff
@@ -33,6 +33,13 @@ export class InvalidContract extends ValidationFailed {
   }
 }
 
+export class InvalidProxy extends ValidationFailed {
+  constructor (proxyHash: string) {
+    super(`Invalid proxy hash: ${proxyHash}`, 'InvalidSignature');
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/utils/errors.ts]: Could be more descriptive, eg.:
`Invalid proxy at address ${proxyAddress}. Contract bytecode hash: ${proxyHash}. Supported hashes: [...]`

File: universal-login-relayer/lib/utils/relayerUnderTest.ts [hunk 2]
```diff
@@ -16,17 +16,22 @@ export class RelayerUnderTest extends Relayer {
...
       chainSpec: {
+        name: 'test',
         ensAddress: ensBuilder.ens.address,
         chainId: 0,
       },
       ensRegistrars: [DOMAIN],
       walletMasterAddress: address,
+      contractWhiteList: {
+        master: [],
+        proxy: ['0x70aa6ef04860e3effad48a2e513965ff76c08c96b7586dfd9e01d4da08e00ccb']
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/utils/relayerUnderTest.ts]: Make dynamic

File: universal-login-relayer/test/helpers/setupTransactionService.js [hunk 3]
```diff
@@ -14,6 +14,6 @@ export default async function setupTransactionService(knex) {
   const pendingMessagesStore = new PendingMessagesSQLStore(knex);
   const transactionQueueStore = new TransactionQueueStore(knex);
   const transactionQueueService = new TransactionQueueService(wallet, provider, transactionQueueStore);
-  const messageHandler = new MessageHandler(wallet, authorisationService, hooks, transactionQueueService, pendingMessagesStore);
+  const messageHandler = new MessageHandler(wallet, authorisationService, hooks, transactionQueueService, pendingMessagesStore, {proxy: ['0x70aa6ef04860e3effad48a2e513965ff76c08c96b7586dfd9e01d4da08e00ccb']});
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/helpers/setupTransactionService.js]: Make dynamic

File: universal-login-relayer/test/helpers/startRelayer.js [hunk 4]
```diff
@@ -25,6 +25,10 @@ async function startRelayer(wallet, relayerConstructor) {
     ensRegistrars: ['mylogin.eth'],
     walletMasterAddress: walletMaster.address,
     tokenContractAddress: tokenContract.address,
+    contractWhiteList: {
+      master: [],
+      proxy: ['0x70aa6ef04860e3effad48a2e513965ff76c08c96b7586dfd9e01d4da08e00ccb']
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/helpers/startRelayer.js]: Dynamic

File: universal-login-relayer/test/relayer/services/messages/MessageValidator.ts [hunk 5]
```diff
@@ -16,7 +16,10 @@ describe('MessageValidator', async () => {
   before(async () => {
     ({mockToken, wallet, walletContract} = await loadFixture(basicWalletContractWithMockToken));
     message = {from: walletContract.address, gasToken: mockToken.address, to: TEST_ACCOUNT_ADDRESS};
-    messageValidator = new MessageValidator(wallet);
+    messageValidator = new MessageValidator(wallet, {
+      master: [],
+      proxy: ['0x70aa6ef04860e3effad48a2e513965ff76c08c96b7586dfd9e01d4da08e00ccb']
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/relayer/services/messages/MessageValidator.ts]: Dynamic

File: universal-login-relayer/lib/config/relayer.ts [hunk 6]
```diff
@@ -30,7 +31,11 @@ const config: Config =  Object.freeze({
     getEnv('ENS_DOMAIN_2', ''),
     getEnv('ENS_DOMAIN_3', ''),
   ],
-  walletMasterAddress: getEnv('WALLET_MASTER_ADDRESS', '')
+  walletMasterAddress: getEnv('WALLET_MASTER_ADDRESS', ''),
+  contractWhiteList: {
+    master: [],
+    proxy: []
```
Comment (marekkirejczyk) [inline universal-login-relayer/lib/config/relayer.ts]: Please state directly supported list of contracts (hash)

File: universal-login-relayer/lib/utils/relayerUnderTest.ts (hunk 2 above)
Comment (xitronix) [reply universal-login-relayer/lib/utils/relayerUnderTest.ts]: ✔️

File: universal-login-relayer/lib/utils/errors.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-relayer/lib/utils/errors.ts]: ✔️

File: universal-login-relayer/test/helpers/setupTransactionService.js (hunk 3 above)
Comment (xitronix) [reply universal-login-relayer/test/helpers/setupTransactionService.js]: ✔️

File: universal-login-relayer/test/helpers/startRelayer.js (hunk 4 above)
Comment (xitronix) [reply universal-login-relayer/test/helpers/startRelayer.js]: ✔️

File: universal-login-relayer/test/relayer/services/messages/MessageValidator.ts (hunk 5 above)
Comment (xitronix) [reply universal-login-relayer/test/relayer/services/messages/MessageValidator.ts]: ✔️

File: universal-login-relayer/lib/config/relayer.ts (hunk 6 above)
Comment (xitronix) [reply universal-login-relayer/lib/config/relayer.ts]: ✔️

File: universal-login-relayer/test/relayer/services/messages/MessageValidator.ts [hunk 7]
```diff
@@ -31,4 +33,14 @@ describe('MessageValidator', async () => {
     await expect(messageValidator.validate(signedMessage, transactionReq))
       .to.be.eventually.rejectedWith('Not enough tokens');
   });
+
+  it('throw error when not correct proxy', async () => {
```
Comment (marekkirejczyk) [inline universal-login-relayer/test/relayer/services/messages/MessageValidator.ts]: Missing test for correct proxy

**Pull request merged**